pkg/instancetype/instancetype.go

```diff
@@ -422,7 +422,7 @@ func (m *InstancetypeMethods) CheckPreferenceRequirements(instancetypeSpec *inst
 	return nil, nil
 }
 
-func (m *InstancetypeMethods) ApplyToVmi(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
+func (m *InstancetypeMethods) ApplyToVmi(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec, vmiMetadata *metav1.ObjectMeta) Conflicts {
```

Comment:
WDYT about changing this func to take a pointer to the VMI instead of taking multiple pointers to multiple fields inside the VMI?

Reply:
I think it would not be possible, because sometimes we are using this function when `VMI` was not created yet and we are passing there the `vm.Spec.Template.Spec`.

Reply:
ACK